### ISSUE TITLE
no-extra-parens produces invalid autofix with an `in` expression in a for-loop initializer

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#11706]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** 12.1.0
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yml
rules:
  no-extra-parens: error
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
for (let a = (b in c); ;);
```

```bash
eslint --fix
```

**What did you expect to happen?**

I expected ESLint to not produce an error because removing the parentheses would cause a syntax error.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint produced the following invalid autofix:

```js
for (let a = b in c; ;);
```

This is interpreted as an initializer in a for-in loop, which is a syntax error in most cases (and is not the intended semantics of the code).

[Demo](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWV4dHJhLXBhcmVuczogZXJyb3IgKi9cblxuZm9yIChsZXQgYSA9IChiIGluIGMpOyA7KTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjksInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

This issue was detected by running the fuzzer locally.

**Are you willing to submit a pull request to fix this bug?**

Yes, but not sure when I'll have time.

**Comment by mdjermanovic** [comment]
Similar bugs, but with parentheses around parent nodes:

```js
for (let a = (b && c in d); ;);

for (let a = (b in c && d); ;);

for (let a = (b => b in c); ;);

for (let a = (b => c => b in c); ;);
```
Question: Disable just autofix or the whole warning? These seem to be necessary parentheses, not 'extra'.

**Comment by not-an-aardvark** [comment]
I suppose the whole warning should be disabled in this case -- as you said, the parentheses are necessary.
